### PR TITLE
Fixed #32219 -- Made InlineModelAdmin.verbose_name_plural fallback to its verbose_name.

### PR DESCRIPTION
**Pull request by Siburg**
Based default verbose_name_plural for an inline admin class on its verbose_name if that is specified.

### CODE REVIEW

File: django/contrib/admin/options.py [hunk 1]
```diff
@@ -2024,10 +2024,13 @@ def __init__(self, parent_model, admin_site):
         self.opts = self.model._meta
         self.has_registered_model = admin_site.is_registered(self.model)
         super().__init__()
+        if self.verbose_name_plural is None:
+            if self.verbose_name is None:
+                self.verbose_name_plural = self.model._meta.verbose_name_plural
+            else:
+                self.verbose_name_plural = self.verbose_name + 's'
```
Comment (pope1ni) [inline django/contrib/admin/options.py:2031]: Make sure you use `format_lazy()` to prevent issues with translated strings:
```suggestion
                self.verbose_name_plural = format_lazy('{}s', self.verbose_name)
```
This is also consistent with the following: https://github.com/django/django/blob/c70cd2a926ffab47f6613e83e0c8828eb6c2c064/django/db/models/options.py#L188-L191

**Review by pope1ni — CHANGES_REQUESTED** [review]

File: django/contrib/admin/options.py (hunk 1 above)
Comment (Siburg) [reply django/contrib/admin/options.py:2031]: Done. I actually wondered about using that, but did not know the reason why. Thank you for pointing that out.

Also took the opportunity to re-order the imports, because that failing isort test was annoying me and at least some of it was my fault.

**Review by smithdc1 — CHANGES_REQUESTED** [review]

File: docs/ref/contrib/admin/index.txt [hunk 2]
```diff
@@ -2400,7 +2400,9 @@ The ``InlineModelAdmin`` class adds or customizes:
 .. attribute:: InlineModelAdmin.verbose_name_plural
 
     An override to the ``verbose_name_plural`` found in the model's inner
-    ``Meta`` class.
+    ``Meta`` class. If a ``verbose_name`` is specified for the inline object
```
[the remark this answers is not in the document]
Comment (Siburg) [reply docs/ref/contrib/admin/index.txt]: Good idea. Done.

File: docs/releases/3.2.txt [hunk 3]
```diff
@@ -511,6 +511,11 @@ backends.
 * Pagination links in the admin are now 1-indexed instead of 0-indexed, i.e.
   the query string for the first page is ``?p=1`` instead of ``?p=0``.
 
+* The default ``verbose_name_plural`` for an ``InlineModelAdmin`` (sub-)class
```
[the remark this answers is not in the document]
Comment (Siburg) [reply docs/releases/3.2.txt]: Done. I was not familiar with that syntax but I hope I have done it correctly.

File: tests/admin_inlines/models.py [hunk 4]
```diff
@@ -337,3 +337,59 @@ class Profile(models.Model):
...
+
+    class Meta:
+        verbose_name_plural = 'Meta Plural Children'
+
+
+class MultiVerboseChild(NonVerboseChild):
+
+    class Meta:
+        verbose_name = 'Meta Multi Child'
+        verbose_name_plural = 'Meta Multi Children'
```
Comment (felixxm) [inline tests/admin_inlines/models.py:395]: Do we need so many new models? It should be possible to reuse `Profile`, e.g.
```python
class VerboseNameProfile(Profile):
    class Meta:
        verbose_name = 'Model verbose name'


class VerboseNamePluralProfile(Profile):
    class Meta:
        verbose_name_plural = 'Model verbose name plural'


class VerboseNamePluralProfile(Profile):
    class Meta:
        verbose_name = 'Model verbose name'
        verbose_name_plural = 'Model verbose name plural'
```
Also all tests which work without this patch should be moved to a separate commit, as additional test coverage.

File: tests/admin_inlines/admin.py [hunk 5]
```diff
@@ -342,6 +344,103 @@ class ClassAdminStackedVertical(admin.ModelAdmin):
...
+
+
+class MultiVerboseParentAdmin(admin.ModelAdmin):
+
+    inlines = [
+        NonVerboseChildMultiVerboseParentInline,
+        VerboseChildMultiVerboseParentInline,
+        VerbosePluralChildMultiVerboseParentInline,
+        MultiVerboseChildMultiVerboseParentInline,
+    ]
```
Comment (felixxm) [inline tests/admin_inlines/admin.py:441]: We can define inlines in the `site.register(..., inlines=[....])`.

File: tests/admin_inlines/admin.py [hunk 6]
```diff
@@ -342,6 +344,103 @@ class ClassAdminStackedVertical(admin.ModelAdmin):
...
+class VerbosePluralChildMultiVerboseParentInline(admin.TabularInline):
+    model = VerbosePluralChild
+    verbose_name = 'Inline Verbose Plural Kid'
+    verbose_name_plural = 'Inline Verbose Plural Kids'
+
+
+class MultiVerboseChildMultiVerboseParentInline(admin.TabularInline):
+    model = MultiVerboseChild
+    verbose_name = 'Inline Multi-Verbose Kid'
+    verbose_name_plural = 'Inline Multi-Verbose Kids'
```
Comment (felixxm) [inline tests/admin_inlines/admin.py:411]: Can we define these classes inside a test case instead? Also all tests which work without this patch should be moved to a separate commit, as additional test coverage.

**Review by felixxm — COMMENTED** [review]
@Siburg Thanks for this patch :+1: I left some comments to tests.

**Comment by Siburg** [comment]
Thank you for your review and comments. I would like to follow up and
clarify with you before I proceed. However, I am on a vacation now and not
in a good position to do what I would like until I get home. I will be back
home on 10 July. If it's OK I will contact you afterwards to ensure I can
then follow your intentions correctly.
Kind regards,
Jelle

**Comment by felixxm** [comment]
@Siburg Sure, take your time :beach_umbrella:

File: tests/admin_inlines/admin.py (hunk 5 above)
Comment (Siburg) [reply tests/admin_inlines/admin.py:441]: Cool. You're right. I did not think of that. Will do.

File: tests/admin_inlines/models.py (hunk 4 above)
Comment (Siburg) [reply tests/admin_inlines/models.py:395]: I'm not sure what you mean by that. Are you proposing a separate PR with tests for only the existing functionality?

File: tests/admin_inlines/admin.py (hunk 6 above)
Comment (Siburg) [reply tests/admin_inlines/admin.py:411]: > Can we define these classes inside a test case instead? 

I gave that a try but couldn't get it to work. It also seems inconsistent with the approach for almost all the other tests in here, so I did not feel like spending more effort on it.

> Also all tests which work without this patch should be moved to a separate commit, as additional test coverage.

Sorry, not sure what you mean by a separate commit. Are you proposing a separate Pull Request to add these tests for the existing functionality of Django 3.2.

Comment (felixxm) [reply tests/admin_inlines/admin.py:411]: > I gave that a try but couldn't get it to work. It also seems inconsistent with the approach for almost all the other tests in here, so I did not feel like spending more effort on it.

OK, I can try to move them with final edits. If a class is only used in the single test, there is no need to define it in "global" scope.

> Sorry, not sure what you mean by a separate commit. Are you proposing a separate Pull Request to add these tests for the existing functionality of Django 3.2.

Yes a separate PR or commit. We don't want to mix bugfixes with extra test coverage.

File: tests/admin_inlines/models.py (hunk 4 above)
Comment (Siburg) [reply tests/admin_inlines/models.py:395]: Well, we want to test 12 combinations, so using 3 types of Parent and 4 types of Child models to achieve that. (We're not testing the combinations where the inlines have neither a `verbose_name` nor a `verbose_name_plural`.) I don't see the benefit of sub-classing a Parent-type model from another test model.

File: tests/admin_inlines/admin.py (hunk 6 above)
Comment (Siburg) [reply tests/admin_inlines/admin.py:411]: Not sure if you actually want to merge this, but a PR with the same tests that reflect the existing functionality in Django 3.2 is here: https://github.com/django/django/pull/14838

**Comment by felixxm** [comment]
@Siburg Thanks :+1: Welcome aboard :boat: 

I simplified tests and pushed small edits to docs.

**Comment by Siburg** [comment]
> @Siburg Thanks +1 Welcome aboard boat
> 
> I simplified tests and pushed small edits to docs.

Thank you for your support and edits. Happy to see it going through.
